app/src/main/java/com/example/sharewhatyoucanproject/donor/AddDonorFragment.kt

```diff
@@ -0,0 +1,200 @@
+package com.example.sharewhatyoucanproject.donor
+
+import android.Manifest
+import android.app.Activity.RESULT_CANCELED
+import android.app.Activity.RESULT_OK
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.location.Location
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import androidx.annotation.RequiresPermission
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
+import com.example.sharewhatyoucanproject.utils.PERMISSION_ALL
+import com.example.sharewhatyoucanproject.utils.foodTypes
+import com.example.sharewhatyoucanproject.utils.hasPermissions
+import com.example.sharewhatyoucanproject.utils.isGPSEnabled
+import com.example.sharewhatyoucanproject.utils.locationPermissions
+import com.example.sharewhatyoucanproject.utils.showToast
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationServices
+import com.google.android.material.progressindicator.CircularProgressIndicator
+import com.google.firebase.firestore.GeoPoint
+import coil.load
+import com.example.sharewhatyoucanproject.AddDonorViewModel
+import com.example.sharewhatyoucanproject.AddDonorViewModelFactory
+import com.example.sharewhatyoucanproject.DonorResult
+import com.example.sharewhatyoucanproject.databinding.FragmentAddDonorBinding
+
+class AddDonorFragment : Fragment()
+{
+    private var _binding: FragmentAddDonorBinding? = null
+    val binding get() = _binding!!
+
+    lateinit var addDonorViewModel: AddDonorViewModel
+    lateinit var locationRequest: LocationRequest
+    lateinit var arrayAdapter: ArrayAdapter<String>
+
+    lateinit var circularProgressIndicator: CircularProgressIndicator
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        addDonorViewModel = ViewModelProvider(
+            this,
+            AddDonorViewModelFactory(),
+        )[AddDonorViewModel::class.java]
+    }
+
+        override fun onCreateView(
+            inflater: LayoutInflater, container: ViewGroup?,
+            savedInstanceState: Bundle?,
+        ): View? {
+            _binding = FragmentAddDonorBinding.inflate(inflater, container, false)
+
+            circularProgressIndicator = binding.progressCircular
+            locationRequest = LocationRequest.create()
+            (activity as AppCompatActivity?)!!.supportActionBar!!.hide()
```

Comment:
no need for double bangs here. if you do not need a support action bar you should be able to set the theme to no action bar

Reply:
I'm trying to hide the support action bar because I wanted to hide it just for this fragment,(while i'm trying to upload food), not for the entire app.